core/orch_test.go

```diff
@@ -765,9 +764,90 @@ func TestPriceInfo_ReturnsBigRat(t *testing.T) {
 	orch := NewOrchestrator(n)
 
 	priceInfo, err := orch.PriceInfo(ethcommon.Address{})
+	assert.Nil(t, err)
+	assert.Equal(t, expPricePerPixel, priceInfo)
+
+	// basePrice = 10/1, txMultiplier = 100/1 => expPricePerPixel = 1010/100
+	basePrice = big.NewRat(10, 1)
+	n.PriceInfo = basePrice
+	orch = NewOrchestrator(n)
+	expPricePerPixel = new(big.Rat).Mul(basePrice, new(big.Rat).Add(big.NewRat(1, 1), new(big.Rat).Inv(txMultiplier)))
+
+	priceInfo, err = orch.PriceInfo(ethcommon.Address{})
+	assert.Nil(t, err)
+	assert.Equal(t, expPricePerPixel, priceInfo)
+
+	// basePrice = 1/10, txMultiplier = 100 => expPricePerPixel = 101/1000
+	basePrice = big.NewRat(1, 10)
+	n.PriceInfo = basePrice
+	orch = NewOrchestrator(n)
+	expPricePerPixel = new(big.Rat).Mul(basePrice, new(big.Rat).Add(big.NewRat(1, 1), new(big.Rat).Inv(txMultiplier)))
 
+	priceInfo, err = orch.PriceInfo(ethcommon.Address{})
 	assert.Nil(t, err)
 	assert.Equal(t, expPricePerPixel, priceInfo)
+
+	// basePrice = 25/10 , txMultiplier = 100 => expPricePerPixel = 2525/1000
+	basePrice = big.NewRat(25, 10)
+	n.PriceInfo = basePrice
+	orch = NewOrchestrator(n)
+	expPricePerPixel = new(big.Rat).Mul(basePrice, new(big.Rat).Add(big.NewRat(1, 1), new(big.Rat).Inv(txMultiplier)))
+
+	priceInfo, err = orch.PriceInfo(ethcommon.Address{})
+	assert.Nil(t, err)
+	assert.Equal(t, expPricePerPixel, priceInfo)
+
+	// basePrice = 10/1 , txMultiplier = 100/10 => expPricePerPixel = 11
+	basePrice = big.NewRat(10, 1)
+	txMultiplier = big.NewRat(100, 10)
+	n.PriceInfo = basePrice
+	recipient = new(pm.MockRecipient)
+	n.Recipient = recipient
+	recipient.On("TxCostMultiplier", mock.Anything).Return(txMultiplier, nil)
+	orch = NewOrchestrator(n)
+	expPricePerPixel = new(big.Rat).Mul(basePrice, new(big.Rat).Add(big.NewRat(1, 1), new(big.Rat).Inv(txMultiplier)))
+
+	priceInfo, err = orch.PriceInfo(ethcommon.Address{})
+	assert.Nil(t, err)
+	assert.Equal(t, expPricePerPixel, priceInfo)
+
+	// basePrice = 10/1 , txMultiplier = 1/10 => expPricePerPixel = 100
```

Comment:
Should `expPricePerPixel = 110`?

`(10 / 1) * (1 + inv(1/ 10)) = (10 / 1) * (1 + 10) = 10 * 11 = 110`

Reply:
Woops, you're right , I forgot doing +1 in my notebook for that one.